### PR TITLE
Board integration

### PR DESCRIPTION
**Pull request by PeterSzadurski**
### What this PR does (required):
- Board is now fully integrated with the backend.
- User now has an array of boards, adding multiboards will be easy from now on
### Screenshots / Videos (required):
- video is a little long due to switching between browser and db)
- it has also been compressed to fit in the pr

https://user-images.githubusercontent.com/44846197/129811124-4508402f-c0ae-4c08-8eaa-1c36c47805ee.mp4

### Any information needed to test this feature (required):
- a new user must be used for this feature, any old users do not have boards assigned to them, the app will crash when using an older user
### Any issues with the current functionality (optional):
- a lot of redundant code since I didn't want to break the unit tests
- authentication/validation needs to be added to the new server code

### CODE REVIEW

File: client/src/context/useAuthBoardContext.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+import logoutAPI from '../helpers/APICalls/logout';
+import { UserBoard } from '../interface/UserBoard';
+import { BoardApi, GetAllBoard } from '../helpers/APICalls/board';
+import { CompleteBoard } from '../interface/BoardApi';
+
+export interface IAuthContext {
+  loggedInUserBoard: UserBoard | null | undefined;
+  updateLoginContext: (data: AuthBoardApiDataSuccess) => void;
+  logout: () => void;
+}
```
Comment (moffatethan) [inline client/src/context/useAuthBoardContext.tsx]: This should be in interface folder

File: client/src/helpers/APICalls/board.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import { Board, Boards, CompleteBoard } from '../../interface/BoardApi';
+import { Column } from '../../interface/ColumnApi';
+
+export interface IBoardContext {
+  board: Board | null | undefined;
+  updateBoardContexts: (data: any) => void;
+}
+
+export const BoardApi = async (): Promise<Boards> => {
+  const boardId = '6114745022197a8d30dc040c';
```
Comment (moffatethan) [inline client/src/helpers/APICalls/board.tsx]: This is constant, we should make it fetch dynamically

File: client/src/helpers/APICalls/board.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+    [],
+  );
+
+  return <BoardContext.Provider value={{ board, updateBoardContexts }}>{children}</BoardContext.Provider>;
+};
+
+export function useBoard(): IBoardContext {
+  return useContext(BoardContext);
+}
```
Comment (moffatethan) [inline client/src/helpers/APICalls/board.tsx]: How come there is a context in here?

File: server/controllers/boards.js [hunk 4]
```diff
@@ -41,6 +42,57 @@ exports.getBoard = asyncHandler(async (req, res) => {
...
+      const doc = await Card.findById(id);
+      console.log("doc: " + doc)
+      cards[doc._id] = doc;
+
+    })));
+
+    console.log("card" + cards);
+
+  }));
+  */
```
Comment (moffatethan) [inline server/controllers/boards.js]: Remove?

File: server/controllers/boards.js [hunk 5]
```diff
@@ -41,6 +42,57 @@ exports.getBoard = asyncHandler(async (req, res) => {
...
+  
+
+  const board = await Board.findById(boardId);
+  /*const columns = await Promise.all(board.columns.map(async (id) => {
+    let column = {};
+    const doc = await Column.findById(id);
+    column[doc._id] = doc;
+    return column;
+  }));
+  */
```
Comment (moffatethan) [inline server/controllers/boards.js]: If we aren't using this, we don't need it in production.

File: server/controllers/boards.js [hunk 6]
```diff
@@ -41,6 +42,57 @@ exports.getBoard = asyncHandler(async (req, res) => {
...
+
+
+  if (board && columns && cards) {
+    res.status(200).json({ board, columns, cards });
+
+  } else {
+    const error = new Error(`Could not find board ${boardId}`);
+    res.status(404).json({
+      error
+    })
```
Comment (moffatethan) [inline server/controllers/boards.js:66]: You could set the status of the response then throw an error and it express-async-handler will respond with a error json response

**Review by moffatethan — COMMENTED** [review]
Left you some comments to address, otherwise, decent work!

File: server/controllers/boards.js (hunk 6 above)
Comment (PeterSzadurski) [reply server/controllers/boards.js:66]:
```
  const error = new Error(`Could not find card ${cardId}`);
  res.status(404).error({
    error
  })
}
```
Would this be correct?

**Pull request merged**